packages/xo-server/src/xo-mixins/pool.mjs

```diff
@@ -30,6 +35,77 @@ export default class Pools {
       sourcePatches[sourceId] = sourceHost.patches
     }
 
+    const hasLinstorSr = some(_app.objects.all, { SR_type: 'linstor', $pool: target.uuid })
+    if (hasLinstorSr) {
+      let nNewHosts = 0
+      let nHostWithoutXcpLicense = 0
+      const xcpLicenses = await _app.getLicenses({ productType: 'xcpng' })
+      const licenseByHostId = keyBy(xcpLicenses, 'boundObjectId')
+      const now = Date.now()
+
+      for (const hostId of sourceIds) {
+        nNewHosts++
+        if (licenseByHostId[hostId] === undefined) {
+          nHostWithoutXcpLicense++
+        }
+      }
+
+      if (nHostWithoutXcpLicense !== 0) {
+        const availableXcpLicenses = xcpLicenses.filter(
+          ({ boundObjectId, expires }) => boundObjectId === undefined && (expires === undefined || expires > now)
+        )
+        const nAvailableXcpLicenses = availableXcpLicenses.length
+        if (nHostWithoutXcpLicense > nAvailableXcpLicenses) {
+          throw new Error(
+            `Not enough XCP Pro Licenses. Expected: ${nHostWithoutXcpLicense}, actual: ${nAvailableXcpLicenses}`
+          )
+        }
+
+        await asyncEach(
+          sourceIds,
+          async hostId => {
+            const xcpLicense = availableXcpLicenses.pop()
+            await _app.bindLicense({
+              licenseId: xcpLicense.id,
+              boundObjectId: hostId,
+            })
+            $defer.onFailure(() =>
+              _app.unbindLicense({ licenseId: xcpLicense.id, productId: xcpLicense.productId, boundObjectId: hostId })
+            )
+          },
+          {
+            stopOnError: false,
+          }
+        )
+      }
+
+      const xostorLicenses = await _app.getLicenses({ productType: 'xostor' })
+      const availableXostorLicenses = xostorLicenses.filter(
+        ({ boundObjectId, expires }) => boundObjectId === undefined && (expires === undefined || expires > now)
+      )
+      const nAvailableXostorLicenses = availableXostorLicenses.length
+      if (nNewHosts > nAvailableXostorLicenses) {
+        throw new Error(`Not enough XOSTOR Licenses. Expected: ${nNewHosts}, actual: ${nAvailableXostorLicenses}`)
+      }
+
+      await asyncEach(
+        sourceIds,
+        async hostId => {
+          const xostorLicense = availableXostorLicenses.pop()
+          await _app.bindLicense({
+            licenseId: xostorLicense.id,
+            boundObjectId: hostId,
+          })
+          $defer.onFailure(() =>
+            _app.unbindLicense({ licenseId: xostorLicense.id, productId: 'xostor', boundObjectId: hostId })
+          )
+        },
+        {
+          stopOnError: false,
+        }
+      )
+    }
```

Comment:
I think this code could be factorized with the code before, maybe  by creating a helper that take a host + license type, compute the host missing the license,  check if enough license are available + bind them